### PR TITLE
UPP-1521: Story package - Create new content-collection-rw-neo4j

### PR DESCRIPTION
**Pull request by ioanaciont**

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+machine:
+  services:
+    - docker
```
Comment (tosan88) [inline circle.yml]: win line endings...

**Comment by tosan88** [comment]
@david-smith-ft We had this discussion, but this service (and other "content-collection-..." services) will be used not just for Curation (or Story package), but Content Packages as well, which are not Curations

File: collection/service.go [hunk 2]
```diff
@@ -0,0 +1,209 @@
+package collection
+
+import (
+	"encoding/json"
+	"github.com/Financial-Times/neo-utils-go/neoutils"
+	//	log "github.com/Sirupsen/logrus"
```
Comment (bogdanguranda) [inline collection/service.go]: You should remove this commented import.

File: collection/service_test.go [hunk 3]
```diff
@@ -0,0 +1,192 @@
+package collection
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	//	"github.com/Financial-Times/base-ft-rw-app-go/baseftrwapp"
```
Comment (bogdanguranda) [inline collection/service_test.go]: Same, we shouldn't keep commented imports.

File: collection/service_test.go [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+		},
+		Result: &result,
+	}
+
+	err := s.conn.CypherBatch([]*neoism.CypherQuery{query})
+
+	return result, err
+}
+
+func getCurationItemsById(uuid string, s service) ([]curationItem, error) {
```
Comment (bogdanguranda) [inline collection/service_test.go]: you can remove uuid param, it is not used in this method, you are using the global var storyPackageUuid

File: collection/service_test.go [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+	}
+
+	conf := neoutils.DefaultConnectionConfig()
+	conf.Transactional = false
+	db, err := neoutils.Connect(url, conf)
+	assert.NoError(err, "Failed to connect to Neo4j")
+	return db
+}
+
+func cleanDB(db neoutils.CypherRunner, t *testing.T, assert *assert.Assertions) {
```
Comment (bogdanguranda) [inline collection/service_test.go]: t param is not used, you can remove it

File: puppet/Modulefile [hunk 6]
```diff
@@ -0,0 +1,13 @@
+name    'ft-content_rw_neo4j'
```
Comment (tosan88) [inline puppet/Modulefile]: Can you please delete the whole puppet folder? We don't use UCS anymore where we needed that

File: main.go [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+	})
+
+	logMetrics := app.Bool(cli.BoolOpt{
+		Name:   "logMetrics",
+		Value:  false,
+		Desc:   "Whether to log metrics. Set to true if running locally and you want metrics output",
+		EnvVar: "LOG_METRICS",
+	})
+
+	env := app.String(cli.StringOpt{
```
Comment (tosan88) [inline main.go]: This is not needed anymore, as this was used only in UCS world

File: main.go [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+		conf.BatchSize = *batchSize
+		db, err := neoutils.Connect(*neoURL, conf)
+
+		if err != nil {
+			log.Errorf("Could not connect to neo4j, error=[%s]\n", err)
+		}
+
+		baseftrwapp.OutputMetricsIfRequired(*graphiteTCPAddress, *graphitePrefix, *logMetrics)
+
+		if *env != "local" {
```
Comment (tosan88) [inline main.go]: This logic also can be removed and used "local" wherever `env` was needed

File: main.go [hunk 9]
```diff
@@ -0,0 +1,156 @@
...
+	m.HandleFunc(status.BuildInfoPathDW, status.BuildInfoHandler)
+
+	m.HandleFunc(status.GTGPath, status.NewGoodToGoHandler(gtg.FailFastParallelCheck(gtgChecker)))
+	return m
+}
+
+func makeCheck(cr neoutils.CypherRunner) v1a.Check {
+	return v1a.Check{
+		BusinessImpact:   "Cannot read/write content via this writer",
+		Name:             "Check connectivity to Neo4j - neoUrl is a parameter in hieradata for this service",
```
Comment (tosan88) [inline main.go]: No hieradata anymore as no puppet anymore

File: main.go [hunk 10]
```diff
@@ -0,0 +1,156 @@
...
+		PanicGuide:       "TODO - write panic guide",
```
Comment (tosan88) [inline main.go]: Can you please create a basic service run book and link that here? e.g. `https://dewey.ft.com/upp-content-rw-neo4j.html`

File: main.go [hunk 11]
```diff
@@ -0,0 +1,139 @@
...
+		db, err := neoutils.Connect(*neoURL, conf)
+
+		if err != nil {
+			log.Errorf("Could not connect to neo4j, error=[%s]\n", err)
+		}
+
+		baseftrwapp.OutputMetricsIfRequired(*graphiteTCPAddress, *graphitePrefix, *logMetrics)
+
+		var m http.Handler
+		m = router(db)
```
Comment (Nesze) [inline main.go]: The rest of the rw-s leverage on the [baseftrwapp](https://github.com/Financial-Times/content-rw-neo4j/blob/master/main.go#L84) to configure the routing, which beside adds some generic endpoints and optional routing, etc.

Couldn't we reuse the baseftrwapp here also?

Comment (ioanaciont) [inline main.go]: Not really, because we need to send params as contentType, relationType for each method (Read, Write, etc.)

Comment (Nesze) [inline main.go]: Right, makes sense.

Though, I think it's worth adding request logging.

File: collection/service.go [hunk 12]
```diff
@@ -0,0 +1,204 @@
...
+	return contentCollectionResult, true, nil
+}
+
+//Write - Writes a content collection node
+func (pcd service) Write(thing interface{}, collectionType string) error {
+	newContentCollection := thing.(contentCollection)
+
+	deleteRelationshipsQuery := &neoism.CypherQuery{
+		Statement: `MATCH (n:Thing {uuid: {uuid}})
+			MATCH (item:Thing)<-[rel:SELECTS]-(n)
```
Comment (tosan88) [inline collection/service.go]: I would use `OPTIONAL MATCH` to include Curations which don't have any relationship

File: collection/service_test.go [hunk 13]
```diff
@@ -0,0 +1,190 @@
...
+
+	err := db.CypherBatch(qs)
+	assert.NoError(err)
+}
+
+func checkDbClean(db neoutils.CypherRunner, t *testing.T) {
+	assert := assert.New(t)
+
+	result := []struct {
+		Uuid string `json:"org.uuid"`
```
Comment (tosan88) [inline collection/service_test.go]: Could you please remove reference to org? If you leave just `uuid` an alias could be used in the CYPHER query

**Pull request merged**